File: wsc/wsc/validations/fee_schedule.py
import frappe
from wsc.wsc.utils import duplicate_row_validation

def validate(doc, method):
    validate_academic_year(doc)
    validate_semester(doc)
    duplicate_row_validation(doc, "student_groups", ['student_group',])

def validate_academic_year(doc):
    if doc.academic_term:
        if doc.academic_term not in [d.name for d in frappe.get_all("Academic Term", {'academic_year':doc.get('academic_year')},['name'])]:
            frappe.throw("Academic Term <b>'{0}'</b> not belongs to academic year <b>'{1}'</b>".format(doc.get('academic_term'), doc.get('academic_year')))

def validate_semester(doc):
    if doc.program not in [d.semesters for d in frappe.get_all("Semesters", {'parent':doc.get('programs')},['semesters'])]:
        frappe.throw("Semester <b>'{0}'</b> not belongs to program <b>'{1}'</b>".format(doc.get('program'), doc.get('programs')))

File: wsc/public/js/fee_schedule.js
frappe.ui.form.on('Fee Schedule',{
    setup: function(frm) {
        frm.set_query("academic_term", function() {
            return {
                filters: {
                    "academic_year":frm.doc.academic_year
                }
            };
        });
        frm.set_query("program", function() {
            return {
                filters: {
                    "programs":frm.doc.programs
                }
            };
        });
    },
    refresh:function(frm){
        frm.set_query("student_group","student_groups", function() {
            return {
                filters: {
                    "program":frm.doc.program
                }
            };
        });
    }
});

frappe.ui.form.on('Fee Schedule',{
    setup: function(frm) {
        frm.set_query("academic_term", function() {
            return {
                filters: {
                    "academic_year":frm.doc.academic_year
                }
            };
        });
        frm.set_query("program", function() {
            return {
                filters: {
                    "programs":frm.doc.programs
                }
            };
        });
    },
    refresh:function(frm){
        frm.set_query("student_group","student_groups", function() {
            return {
                filters: {
                    "program":frm.doc.program
                }
            };
        });
        if (frm.doc.fee_creation_status === 'Successful') {
			frm.add_custom_button(__('Accounting Ledger'), function() {
				frappe.route_options = {
					// voucher_no: frm.doc.name,
                    voucher_no: frm.doc.name,
					from_date: frm.doc.posting_date,
					to_date: moment(frm.doc.modified).format('YYYY-MM-DD'),
					company: frm.doc.company,
					group_by: '',
					show_cancelled_entries: frm.doc.docstatus === 2
				};
				frappe.set_route("query-report", "Fee structure General Ledger");
                // frappe.set_route("query-report", "General Ledger");
                
			});
		}
    }
});

frappe.ui.form.on('Fee Schedule', {
    refresh:function(frm) {
		if(frappe.user.has_role(["Accounts User","Student","Education Administrator"]) && !frappe.user.has_role(["Administrator"])){
  			frm.remove_custom_button('Create Fees');
        }
	}
}
);